Add Profiles concepts page (#9339)
open-telemetry/opentelemetry.io: content/en/docs/concepts/signals/profiles.md

- **Sampling-based profiling**: A profiler periodically interrupts the program
  (e.g. using timer-based interrupts) and records the current stack trace. This
  is the most common approach for CPU profiling. On Linux, profilers can use
  eBPF to capture stack traces from the kernel without modifying userspace
  applications at all. This approach enables zero-instrumentation, whole-system
  profiling (including code produced by compiled languages without runtime
  support) and is designed for continuous, low-overhead production use.
- **Instrumentation-based profiling**: Runtime hooks or bytecode instrumentation
  report events like memory allocations, lock acquisitions or garbage
  collections along with their associated stack traces.

Regardless of the collection method, the resulting data is serialized into
OpenTelemetry's common profile data model and exported via OTLP.

## Collecting profiles

OpenTelemetry provides an
[eBPF-based profiling agent](https://github.com/open-telemetry/opentelemetry-ebpf-profiler)
for Linux, capable of profiling most languages with zero code changes.

Additional language-specific profiling integrations that tap into built-in

> runtime profiling frameworks (e.g. JFR for Java, pprof for Go) will also become

Reviewer: ```suggestion
runtime profiling frameworks, such as JFR for Java or pprof for Go, will also become
```
